### DISCUSSION TITLE
How to correctly average dB with pint

### DISCUSSION THREAD

**Discussion by spirrobe** [hgrecco/pint#1999] [Q&A]
Hi

I'm considering using pint for a project of mine that also uses xarray. 
One issue is the resampling/grouping/statistics (which should happen in lin space), as values are usually reported in dB.
While I can deal with this myself (and there is some logic to it that relies on attributes and xarray accessors), it would be better to keep dB as quantity linked to the values - which would allow me to not rely on attributes which depending on the operation are not passed forward.

However, looking at the log units which are per [related doc entry](https://pint.readthedocs.io/en/stable/user/log_units.html) certainly in beta I face the issue that for calculations I'd need to downcast beforehand nonetheless (resp. dB or log units are treated as linear units).
MWE:
```python3

import numpy as np
from pint import UnitRegistry

ureg_autoconv = UnitRegistry(autoconvert_offset_to_baseunit=True)
ureg_noautoconv = UnitRegistry(autoconvert_offset_to_baseunit=False)
basearr = np.asarray([30, 20])
a = basearr * ureg_noautoconv.dB
b = basearr * ureg_autoconv.dB
c = 10**(basearr/10)

print('not correct (no autoconversion)', np.mean(a), a.mean(), '\n',
      'not correct (autoconversion)', np.mean(b), b.mean(), '\n',
      'correct result when converting to linear space first', 10*np.log10(c.mean()))
# output
# not correct (no autoconversion) 25.0 decibel 25.0 decibel 
#  not correct (autoconversion) 25.0 decibel 25.0 decibel 
#  correct result when converting to linear space first 27.403626894942438
```

Is there a proper way to handle this? 
Will the log unit feature include such averaging (or other similar functions like sum, std, for that matter) when it's out of beta or should I always assume that I have to convert myself?

**Comment by andrewgsavage** [comment]
I'd expect all of the numpy array funcs to behave like that for a while - until someone puts in a PR adding support for them. 

I think for those functions the `pre_calc_units` https://github.com/hgrecco/pint/blob/9972e60ead959f9ecfb844dd1ab552b2b20477bc/pint/facets/numpy/numpy_func.py#L306 (may also be in other places) needs to be set to the linear unit for the given log unit.
The result unit should already be set to be the same unit as the input
